sdk/identity/identity-web/src/index.ts

```diff
@@ -0,0 +1,207 @@
+  private clientId: string;
+  private redirectUri: string;
+  private credentialOptions: WebRedirectCredentialOptions;
+  private msalFlow: MsalFlow | undefined;
+
+  /**
+   * Creates an instance of {@link WebRedirectCredential}.
+   *
+   * ```javascript
+   * const credential = new WebRedirectCredential(
+   *   tenantId,
+   *   clientId,
+   *   redirectUri
+   * );
+   * ```
+   *
+   * This credential won't automatically authenticate at first.
+   * It is necessary to go through the redirection process to retrieve a code:
+   *
+   * ```javascript
+   * const authorizeUrl = credential.getRedirectUri(scope);
+   * // Redirect to the authorizeUrl
+   * ```
+   *
+   * After the redirection, the redirect request will include a `code`.
+   * With this code, we can authenticate the first time using `authenticate`:
+   *
+   * ```javascript
+   * const authenticationRecord = await credential.authenticate(scope, code);
+   * ```
+   *
+   * Once we have authenticated, subsequent authentication requests will work.
+   * The `authorizationRecord` can be stored and retrieved for later use:
+   *
+   * ```javascript
+   * const credential2 = new WebRedirectCredential(
+   *   tenantId,
+   *   clientId,
+   *   redirectUri,
+   *   { authenticationRecord }
+   * );
+   * ```
+   *
+   * @param tenantId - The Azure Active Directory tenant (directory) ID or name.
+   * @param clientId - The client (application) ID of an App Registration in the tenant.
+   * @param redirectUri - The redirect URI that was used to request the authorization code.
+   * @param options - Options for configuring the client which makes the access token request.
+   */
+  constructor(
+    tenantId: string,
+    clientId: string,
+    redirectUri: string,
+    options: WebRedirectCredentialOptions = {}
+  ) {
+    checkTenantId(logger, tenantId);
+
+    this.tenantId = tenantId;
+    this.clientId = clientId;
+    this.redirectUri = redirectUri;
+    this.credentialOptions = options;
+  }
+
+  /**
+   * Generates the Authorize URI
+   * where the application must redirect to
+   * in order to trigger the Authorization Code Flow.
+   */
+  getAuthorizeUrl(
+    scopes: string,
+    options: {
+      /**
+       * Helps customize the authorization host, in case of testing or private clouds.
+       */
+      authorizeHost?: string;
+      /**
+       * Metadata useful to know the source of the redirection.
+       */
+      state?: string;
+    } = {}
+  ): string {
+    let queryParams: Record<string, string | undefined> = {
+      client_id: this.clientId,
+      response_type: "code",
+      redirect_uri: this.redirectUri,
+      scope: scopes,
+    };
+    if (options.state) {
+      queryParams.state = options.state;
+    }
+    const params = new URLSearchParams(queryParams as Record<string, string>);
+    const query = params.toString();
+    const authorizeHost = `https://login.microsoftonline.com/${this.tenantId}/oauth2/v2.0/authorize?${query}`;
```

Comment:
This `login.microsoft.com` most likely changes depending on the cloud.

Feedback from @KarishmaGhiya : Use the authority host.

Reply:
Because `getAuthorizeUrl(` is part of a credential class, and we’re re-using the code from the main identity package, we can derive the authority host from the MSAL internals, we just need to re-use the code we use there.